### ISSUE TITLE
 spine 1.2.12 does not draw a graph

### ISSUE THREAD

**Issue by bredis** [Cacti/spine#157]
Installed cacti 1.2.12 and compiled spine 1.2.12 to debian 10. rrdtool 1.7.1.
The problem is exactly the same as I described in this question https://github.com/Cacti/spine/issues/121.
On a clean installation, he created Data Input Methods, Data Source and Graph. Poller interval, cron, Data Source Profile is 1 minute.
An rrd file was created, but no data is added to it. When poller changes to cmd.php, the graph starts to draw.

**Comment by netniV** [comment]
Did you `chmod +s spine` and `chown root:root spine` ?  Try running spine manually in debug and/or readonly mode for the device you are having an issue with.

**Comment by bredis** [comment]
I do chmod +s spine and chown root:root spine.
`root@ns:~# php -q /var/www/cacti/poller.php --debug
2020-05-24 15:40:04 - POLLER: Poller[1] NOTE: Poller Int: '60', Cron Int: '60', Time Since Last: '3.45', Max Runtime '58', Poller Runs: '1'
2020-05-24 15:40:04 - POLLER: Poller[1] NOTE: Cron is configured to run too often!  The Poller Interval is '60' seconds, with a minimum Cron period of '60' seconds, but only 3.5 seconds have passed since the poller last ran.
root@ns:~# php -q /var/www/cacti/poller.php --debug
2020-05-24 15:40:16 - POLLER: Poller[1] NOTE: Poller Int: '60', Cron Int: '60', Time Since Last: '15.36', Max Runtime '58', Poller Runs: '1'
2020-05-24 15:40:16 - POLLER: Poller[1] NOTE: Cron is configured to run too often!  The Poller Interval is '60' seconds, with a minimum Cron period of '60' seconds, but only 15.4 seconds have passed since the poller last ran.
root@ns:~# php -q /var/www/cacti/poller.php --debug --force
2020-05-24 15:40:28 - POLLER: Poller[1] NOTE: Poller Int: '60', Cron Int: '60', Time Since Last: '27.87', Max Runtime '58', Poller Runs: '1'
2020-05-24 15:40:28 - POLLER: Poller[1] DEBUG: About to Spawn a Remote Process [CMD: /usr/local/spine/bin/spine, ARGS:  -C '/usr/local/spine/etc/spine.conf' --poller=1 --first=0 --last=0 --mibs]
2020-05-24 15:40:28 - POLLER: Poller[1] DEBUG: About to Spawn a Remote Process [CMD: /usr/local/spine/bin/spine, ARGS:  -C '/usr/local/spine/etc/spine.conf' --poller=1 --first=2 --last=2 --mibs]
Waiting on 1 of 2 pollers.
OK u:0,00 s:0,00 r:0,99
2020-05-24 15:40:30 - SYSTEM STATS: Time:1.2250 Method:spine Processes:1 Threads:1 Hosts:1 HostsPerProcess:1 DataSources:1 RRDsProcessed:1
2020-05-24 15:40:30 - POLLER: Poller[1] DEBUG: About to Spawn a Remote Process [CMD: /usr/bin/php, ARGS: -q /var/www/cacti/poller_reports.php]
2020-05-24 15:40:30 - POLLER: Poller[1] DEBUG: About to Spawn a Remote Process [CMD: /usr/bin/php, ARGS: -q '/var/www/cacti/poller_spikekill.php']
2020-05-24 15:40:30 - POLLER: Poller[1] DEBUG: About to Spawn a Remote Process [CMD: /usr/bin/php, ARGS:  -q '/var/www/cacti/poller_automation.php' -M]
2020-05-24 15:40:30 - POLLER: Poller[1] DEBUG: About to Spawn a Remote Process [CMD: /usr/bin/php, ARGS:  -q '/var/www/cacti/poller_maintenance.php']
root@ns:~# php -q /var/www/cacti/poller.php --debug
2020-05-24 15:40:37 - POLLER: Poller[1] NOTE: Poller Int: '60', Cron Int: '60', Time Since Last: '9.22', Max Runtime '58', Poller Runs: '1'
2020-05-24 15:40:37 - POLLER: Poller[1] NOTE: Cron is configured to run too often!  The Poller Interval is '60' seconds, with a minimum Cron period of '60' seconds, but only 9.2 seconds have passed since the poller last ran.
root@ns:~#
`

**Comment by netniV** [comment]
If you snmpwalk that device, which output do you see, spine or cmd?

**Comment by bredis** [comment]
```
root@ns:~# snmpwalk -v2c -c public 192.168.31.3:10161 .1.3.6.1.4.1.2021.8.1.101.2
UCD-SNMP-MIB::extOutput.2 = STRING: CONNECTIONS:19
```
correct output from cmd.php

**Comment by anarkia1976** [comment]
Hi @netniV , i have reverted git commits to make spine working again:

git rh af33bbc5ecb5d0864853478b12b990c29fb5a393

Now after clean compile from source spine is ok.

Thanks a lot.

**Comment by netniV** [comment]
Was that for this issue?

**Comment by anarkia1976** [comment]
> Was that for this issue?

I'm switching to a new issue ticket. thanks.

**Comment by TheWitness** [comment]
There was an issue corrected since the 1.2.12 release was issued.  This fix is here.

https://github.com/Cacti/spine/commit/3d0adde5dc90656d5dc692303de84689c5b21eb3

**Comment by bredis** [comment]
I compiled spine 1.2.13, but the double quote problem still persists.
``` root@ns:~/cacti-spine-1.2.13# /usr/local/spine/bin/spine -v
SPINE 1.2.13  Copyright 2004-2020 by The Cacti Group
root@ns:~/cacti-spine-1.2.13#
```
``` 
2020-07-29 17:53:01 - SPINE: Poller[Main Poller] PID[8525] DEVDBG: SQL:INSERT INTO poller_output (local_data_id, rrd_name, time, output) VALUES (52,'Connections','2020-07-29 17:53:01','"CONNECTIONS:6') ON DUPLICATE KEY UPDATE output=VALUES(output)
--
2020-07-29 17:53:01 - SPINE: Poller[Main Poller] PID[8525] Device[192.168.31.3] HT[1] DS[192.168.31.3 - Keenetic - Connections] Graphs[192.168.31.3 - Keenetic - Connections] SCRIPT: /usr/bin/php /var/www/cacti/scripts/query_keenetic_connections.php '192.168.31.3' '10161' 'public' , output: "CONNECTIONS:6
```

Output = `"CONNECTIONS:6`
